sdk/compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Upgrade.hs

```diff
@@ -58,22 +61,40 @@ runGammaUnderUpgrades Upgrading{ _past = pastAction, _present = presentAction }
     presentResult <- withReaderT _present presentAction
     pure Upgrading { _past = pastResult, _present = presentResult }
 
-checkUpgrade :: Version -> Upgrading LF.Package -> [Diagnostic]
-checkUpgrade version package =
-    let upgradingWorld = fmap (\package -> emptyGamma (initWorldSelf [] package) version) package
-        result =
-            runGammaF
-                upgradingWorld
-                (checkUpgradeM package)
+checkUpgrade :: Version -> Bool -> LF.Package -> Maybe (LF.PackageId, LF.Package) -> [Diagnostic]
+checkUpgrade version shouldTypecheckUpgrades presentPkg mbUpgradedPackage =
+    let bothPkgDiagnostics :: Either Error ((), [Warning])
+        bothPkgDiagnostics =
+            case mbUpgradedPackage of
+                Nothing ->
+                    Right ((), [])
+                Just (_, pastPkg) ->
+                    let package = Upgrading { _past = pastPkg, _present = presentPkg }
+                        upgradingWorld = fmap (\package -> emptyGamma (initWorldSelf [] package) version) package
+                    in
+                    runGammaF upgradingWorld $ do
+                        when shouldTypecheckUpgrades (checkUpgradeM package)
+
+        singlePkgDiagnostics :: Either Error ((), [Warning])
+        singlePkgDiagnostics =
+            let world = initWorldSelf [] presentPkg
+            in
+            runGamma world version $ do
+                 checkNewInterfacesHaveNoTemplates presentPkg
+                 checkNewInterfacesAreUnused presentPkg
```

Comment:
Decide which of these error messages are more important and prioritize it in the error flags PR

Reply:
233ce273cd